### PR TITLE
resource_control: use override priority if specified

### PR DESCRIPTION
**Pull request by Connor1996**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/14925

What's Changed:

```commit-message
use override priority to replace resource group priority for queries if specified
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by Connor1996** [comment]
PTAL @CabinfeverB

File: components/resource_control/src/resource_group.rs [hunk 1]
```diff
@@ -423,7 +464,12 @@ impl GroupPriorityTracker {
         } else {
             self.virtual_time.load(Ordering::Relaxed)
         }) + task_extra_priority;
-        concat_priority_vt(self.group_priority, vt)
+        let priority = if let Some(p) = override_priority {
+            p
+        } else {
+            self.group_priority
+        };
```
Comment (glorv) [inline components/resource_control/src/resource_group.rs:471]:
```suggestion
        let priority = override_priority.unwrap_or(self.group_priority);
```

File: src/read_pool.rs [hunk 2]
```diff
@@ -120,7 +123,7 @@ impl ReadPoolHandle {
         f: F,
         priority: CommandPri,
         task_id: u64,
-        group_meta: Vec<u8>,
+        resource_control_ctx: &ResourceControlContext,
```
Comment (glorv) [inline src/read_pool.rs:126]: Is it better to use `TaskMetadata` metadata here?

File: src/server/service/kv.rs [hunk 3]
```diff
@@ -197,9 +197,9 @@ macro_rules! handle_request {
             let begin_instant = Instant::now();
 
             let source = req.mut_context().take_request_source();
-            let resource_control_ctx = req.get_context().get_resource_control_context();
+            let resource_control_ctx = req.get_context().get_resource_control_context().clone();
```
Comment (glorv) [inline src/server/service/kv.rs:200]: Why need clone here?

Comment (Connor1996) [reply src/server/service/kv.rs:200]: oops, miss change in global replace

File: components/resource_control/src/resource_group.rs [hunk 4]
```diff
@@ -383,14 +383,55 @@ impl ResourceController {
...
+pub struct TaskMetadata {
+    // the resource group name of this task.
+    pub group_name: String,
+    // the override priority of this task neglecting the original resource group priority.
+    pub override_priority: u32,
+}
+
+impl TaskMetadata {
+    pub fn to_vec(self) -> Vec<u8> {
+        let mut buf = Vec::with_capacity(4 + self.group_name.len());
```
Comment (glorv) [inline components/resource_control/src/resource_group.rs:399]: NIT: maybe we can avoid an extra malloc by reusing the memory of group_name if it already has the extra 4 bytes.

**Review by glorv — APPROVED** [review]
LGTM

**Comment by Connor1996** [comment]
Reduce memory copy cost further, PTAL again @glorv

**Review by glorv — COMMENTED** [review]
LGTM

**Review by nolouch — APPROVED** [review]
lgtm

**Review by tabokie — APPROVED** [review]

**Pull request merged**